set.mm

```diff
@@ -548850,6 +548900,515 @@ class of symmetric relations as well (like the elements of the class of
+$)
+
+  $( Define the class of all transitive sets (vs. the transitive class defined
+     in ~ df-tr ).  It is used only by ~ df-trrels .
+
+     Note the similarity of the definitions of ~ df-refs , ~ df-syms and
+     ~ df-trs .  (Contributed by Peter Mazsa, 17-Jul-2021.) $)
+  df-trs $a |- Trs = { x |
+      ( ( x i^i ( dom x X. ran x ) ) o. ( x i^i ( dom x X. ran x ) ) ) _S
+        ( x i^i ( dom x X. ran x ) ) } $.
+
+  $( Define the class of transitive relations.  The element of the class of
+     transitive relations and the transitive relation predicate are the same
+     when ` R ` is a set, cf. ~ eltrrelsrel .  Alternate definitions are
+     ~ dftrrels2 and ~ dftrrels3 .
+
+     This definition is similar to the definitions of the classes of reflexive
+     ( ~ df-refrels ) and symmetric ( ~ df-symrels ) relations.  (Contributed
+     by Peter Mazsa, 7-Jul-2019.) $)
+  df-trrels $a |- TrRels = ( Trs i^i Rels ) $.
+
+  $( Define the transitive relation predicate.  (Read: ` R ` is a transitive
+     relation.)  The element of the class of all transitive relations and the
+     transitive relation predicate are the same when ` R ` is a set, cf.
+     ~ eltrrelsrel .  Alternate definitions are ~ dftrrel2 and ~ dftrrel3 .
+     (Contributed by Peter Mazsa, 17-Jul-2021.) $)
+  df-trrel $a |- ( TrRel R <->
+      ( ( ( R i^i ( dom R X. ran R ) ) o. ( R i^i ( dom R X. ran R ) ) ) C_
+          ( R i^i ( dom R X. ran R ) ) /\ Rel R ) ) $.
+
+  $( Alternate definition of the class of transitive relations.
+
+     I'd prefer to define the class of transitive relations by using the
+     definition of composition by [Suppes] p. 63. df-coSUP
+     ` ( A o. B ) = { <. x , y >. | E. u ( x A u /\ u B y ) } ` as opposed to
+     the present definition of composition ~ df-co
+     ` ( A o. B ) = { <. x , y >. | E. u ( x B u /\ u A y ) } ` because the
+     Suppes definition keeps the order of ` A ` , ` B ` , ` C ` , ` R ` ,
+     ` S ` , ` T ` by default in trsinxpSUP ( ( ( R i^i ( A X. B ) ) o.  ( S
+     i^i ( B X. C ) ) ) C_ ( T i^i ( A X. C ) ) <-> A. x e.  A A. y e.  B A. z
+     e.  C ( ( x R y /\ y S z ) -> x T z ) ) while the present definition of
+     composition disarranges them: trsinxp ( ( ( S i^i ( B X. C ) ) o.  ( R i^i
+     ( A X. B ) ) ) C_ ( T i^i ( A X. C ) ) <-> A. x e.  A A. y e.  B A. z e.
+     C ( ( x R y /\ y S z ) -> x T z ) ) This is not mission critical to me,
+     the implication of the Suppes definition is just more aesthetic, at least
+     in the above case.
+
+     If we swap to the Suppes definition of class composition, I would define
+     the present class of all transitive sets as df-trsSUP and I would consider
+     to switch the definition of the class of cosets by ` R ` from the present
+     ~ df-coss to a df-cossSUP. But perhaps there is a mathematical reason to
+     keep the present definition of composition.  (Contributed by Peter Mazsa,
```

Comment:
One has (f o g)(x) = f(g(x)). If you swap composition, then to be fully consistent you should also write function application in a postfix way. Some authors do that, e.g., Jacobson, Lectures in Abstract Algebra.
Changing the definition in set.mm would be too much work for unclear benefit.

Reply:
I agree with @benjub not to change the current definition of composition.